libcloud/compute/drivers/vsphere.py

```diff
@@ -0,0 +1,1992 @@
+                    template = vm
+        except Exception as exc:
+            raise LibcloudError("Error while searching for template, ", exc)
+        if not template:
+            raise LibcloudError("Unable to locate VirtualMachine.")
+
+        return template
+
+    def find_by_uuid(self, node_uuid):
+        """Searches VMs for a given uuid
+        returns pyVmomi.VmomiSupport.vim.VirtualMachine
+        """
+        vm = self.connection.content.searchIndex.FindByUuid(None, node_uuid,
+                                                            True, True)
+        if not vm:
+            # perhaps it is a moid
+            vm = self._get_item_by_moid('VirtualMachine', node_uuid)
+            if not vm:
+                raise LibcloudError("Unable to locate VirtualMachine.")
+        return vm
+
+    def find_custom_field_key(self, key_id):
+        """Return custom field key name, provided it's id
+        """
+        if not hasattr(self, "custom_fields"):
+            content = self.connection.RetrieveContent()
+            if content.customFieldsManager:
+                self.custom_fields = content.customFieldsManager.field
+            else:
+                self.custom_fields = []
+        for k in self.custom_fields:
+            if k.key == key_id:
+                return k.name
+        return None
+
+    def get_obj(self, vimtype, name):
+        """
+        Return an object by name, if name is None the
+        first found object is returned
+        """
+        obj = None
+        content = self.connection.RetrieveContent()
+        container = content.viewManager.CreateContainerView(
+            content.rootFolder, vimtype, True)
+        for c in container.view:
+            if name:
+                if c.name == name:
+                    obj = c
+                    break
+            else:
+                obj = c
+                break
+        return obj
+
+    def wait_for_task(self, task, timeout=1800, interval=10):
+        """ wait for a vCenter task to finish """
+        start_time = time.time()
+        task_done = False
+        while not task_done:
+            if (time.time() - start_time >= timeout):
+                raise LibcloudError('Timeout while waiting '
+                                    'for import task Id %s'
+                                    % task.info.id)
+            if task.info.state == 'success':
+                if task.info.result and str(task.info.result) != 'success':
+                    return task.info.result
+                return True
+
+            if task.info.state == 'error':
+                raise LibcloudError(task.info.error.msg)
+            time.sleep(interval)
+
+    def create_node(self, **kwargs):
+        """
+        Creates and returns node.
+
+        :keyword    ex_network: Name of a "Network" to connect the VM to ",
+        :type       ex_network: ``str``
+
+        """
+        name = kwargs['name']
```

Comment:
Please directly specify all the required and possible arguments in the method signature instead of using ``**kwargs``.

This makes it much more user-friendly for the end user.